service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ChildSubmissionController.java

```diff
@@ -45,13 +61,34 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody Callback
     private List<String> validate(CaseDetails caseDetails) {
         ImmutableList.Builder<String> errors = ImmutableList.builder();
 
-        Map<String, Object> childrenData = (Map<String, Object>) caseDetails.getData().get("children");
-        Children children = mapperService.mapObject(childrenData, Children.class);
-        if (children.getAllChildren().stream()
-            .map(Child::getChildDOB)
-            .filter(Objects::nonNull)
-            .anyMatch(dateOfBirth -> dateOfBirth.after(new Date()))) {
-            errors.add("Date of birth cannot be in the future");
+        Map<String, Object> childrenData =
+            (Map<String, Object>) defaultIfNull(caseDetails.getData().get("children"), null);
+
+        if (caseDetails.getData().containsKey("children1")) {
+            List<Map<String, Object>> migratedChildrenObject =
+                (List<Map<String, Object>>) caseDetails.getData().get("children1");
+
+            List<MigratedChildren> migratedChildren = migratedChildrenObject.stream()
+                .map(child ->
+                    mapperService.mapObject((Map<String, Object>) child.get("value"), MigratedChildren.class))
+                .collect(toList());
+
+            if (migratedChildren.stream()
+                .map(MigratedChildren::getParty)
+                .map(Party::getDateOfBirth)
+                .filter(Objects::nonNull)
+                .anyMatch(dateOfBirth -> dateOfBirth.after(new Date()))) {
+                errors.add("Date of birth cannot be in the future");
+            }
+        } else {
+
+            OldChildren children = mapperService.mapObject(childrenData, OldChildren.class);
+            if (children.getAllChildren().stream()
+                .map(OldChild::getChildDOB)
+                .filter(Objects::nonNull)
+                .anyMatch(dateOfBirth -> dateOfBirth.after(new Date()))) {
+                errors.add("Date of birth cannot be in the future");
+            }
```

Comment:
If I hit this with empty caseDetails i.e no children or children1, what happens? Will it hit the else and throw a null pointer?

Reply:
Lets include a test